src/services/dumpers/agent-nodejs.ts

```diff
@@ -0,0 +1,190 @@
+
+  writeIndex(dbDialect: string) {
+    const isMongoose = dbDialect === 'mongodb';
+
+    const context = {
+      isMongoose,
+      isMySQL: dbDialect === 'mysql',
+      isMSSQL: dbDialect === 'mssql',
+      isMariaDB: dbDialect === 'mariadb',
+      forestServerUrl: this.env.FOREST_URL_IS_DEFAULT ? false : this.env.FOREST_SERVER_URL,
+      datasourceCreation: isMongoose
+        ? 'createMongooseDataSource(connection, {})'
+        : `
+    createSqlDataSource({
+      uri: process.env.DATABASE_URL,
+      schema: process.env.DATABASE_SCHEMA || 'public',
+      ...dialectOptions,
+    }),
+  `,
+      datasourceImport: isMongoose
+        ? `const { createMongooseDataSource } = require('@forestadmin/datasource-mongoose');\nconst connection = require('./mongoose-models');`
+        : `const { createSqlDataSource } = require('@forestadmin/datasource-sql');`,
+    };
+
+    this.copyHandleBarsTemplate('index.hbs', 'index.js', context);
+  }
+
+  private writeDotEnv(
+    dbConfig: DbConfigInterface,
+    applicationPort: number,
+    forestEnvSecret: string,
+    forestAuthSecret: string,
+  ) {
+    const databaseUrl = this.buildDatabaseUrl(dbConfig);
+    const context = {
+      databaseUrl,
+      databaseSsl: dbConfig.ssl || false,
+      databaseSchema: dbConfig.dbSchema !== '' ? dbConfig.dbSchema : false,
+      applicationPort,
+      forestServerUrl: this.env.FOREST_URL_IS_DEFAULT ? false : this.env.FOREST_SERVER_URL,
+      forestEnvSecret,
+      forestAuthSecret,
+      hasDockerDatabaseUrl: false,
+      dockerDatabaseUrl: undefined,
+    };
+    if (!this.isLinuxOs) {
+      context.dockerDatabaseUrl = databaseUrl.replace('localhost', 'host.docker.internal');
+      context.hasDockerDatabaseUrl = true;
+    }
+    this.copyHandleBarsTemplate('env.hbs', '.env', context);
+  }
+
+  private writeGitignore() {
+    this.writeFile('.gitignore', 'node_modules\n.env');
+  }
+
+  private writeDockerignore() {
+    this.writeFile('.dockerignore', 'node_modules\nnpm-debug.log\n.env');
+  }
+
+  private writeDockerfile() {
+    this.copyHandleBarsTemplate('Dockerfile.hbs', 'Dockerfile');
+  }
+
+  private writeDockerCompose(config: ConfigInterface) {
+    const databaseUrl = `\${${this.isLinuxOs ? 'DATABASE_URL' : 'DOCKER_DATABASE_URL'}}`;
+    const forestServerUrl = this.env.FOREST_URL_IS_DEFAULT ? false : `\${FOREST_SERVER_URL}`;
+
+    let forestExtraHost: string | boolean = false;
+    if (forestServerUrl) {
+      try {
+        const parsedForestUrl = new URL(this.env.FOREST_SERVER_URL);
+        forestExtraHost = parsedForestUrl.hostname;
+      } catch (error) {
+        throw new Error(`Invalid value for FOREST_SERVER_URL: "${this.env.FOREST_SERVER_URL}"`);
+      }
+    }
+
+    this.copyHandleBarsTemplate('docker-compose.hbs', 'docker-compose.yml', {
+      containerName: this.snakeCase(config.appConfig.applicationName),
+      databaseUrl,
+      dbSchema: config.dbConfig.dbSchema !== '' ? config.dbConfig.dbSchema : false,
```

Comment:
```suggestion
      dbSchema: config.dbConfig.dbSchema ? config.dbConfig.dbSchema : false,
```

Reply:
The default value of the schema is an empty string, so we need to explicitely test it.

Reply:
🔴